### DISCUSSION TITLE
calling to FS.syncfs in Worker for improving rendering frame rate

### DISCUSSION THREAD

**Discussion by aharondavid** [emscripten-core/emscripten#18827] [Q&A]
To improve rendering frame rate on the main thread , I thought to move a call to FS.syncfs from the Main Thread to a Worker thread.
I call to FS.syncfs on every IDBFS file close.
but, is FS.syncfs safe thread? should I lock\unlock calling to  FS.syncfs in worker ?

**Comment by sbc100** [comment]
The emscripten filesytem is implemented on the main thread (at least in its current state, I think WASMFS is changing that). That means that all FS operations are proxied back the main thread.  It also means that accessing the FS object from JavaScript on a worker simply doesn't work.   The FS object (at the least the one exposed to native code) lives on the main thread only.

**Comment by aharondavid** [comment]
Ok, thanks.
